anesthetic/read/cobayamcmcreader.py

```diff
@@ -32,28 +32,21 @@ def paramnames(self):
         except IOError:
             return super().paramnames()
 
-    def limits(self):
-        """Infer param limits from <root>.yaml in cobaya format."""
-        s = loadMCSamples(file_root=self.root)
-        return {p.name: (s.ranges.getLower(p.name), s.ranges.getUpper(p.name))
-                for p in s.paramNames.names
-                if s.ranges.getLower(p.name) is not None
-                or s.ranges.getUpper(p.name) is not None}
-
     def samples(self, burn_in=False):
         """Read <root>_1.txt in getdist format."""
         data = np.array([])
         for chains_file in self.chains_files:
             data_i = np.loadtxt(chains_file)
             if burn_in:
-                if 0 < burn_in < 1:
+                if isinstance(burn_in, float) and 0 < burn_in < 1:
                     index = int(len(data_i) * burn_in)
-                elif type(burn_in) is int and 1 < burn_in < len(data_i):
+                elif isinstance(burn_in, int) and 1 < burn_in < len(data_i):
```

Comment:
Does it need to be an integer? Pandas slicing will probably handle floats, and there's no need to restrict this to users (it would be annoying to have to wrap an `int(...)` around an automatically computed argument. You're also missing out the case where burn_in=1.